esmvalcore/cmor/_fixes/icon/_base_fixes.py

```diff
@@ -27,9 +31,155 @@ def __init__(self, *args, **kwargs):
+        (node_lat, node_lon) = self._get_node_coords(horizontal_grid)
+
+        # The bounds given by the face coordinates slightly differ from the
+        # bounds determined by the connectivity. We arbitrarily assume here
+        # that the information given by the connectivity is correct.
+        conn_node_inds = vertex_of_cell.data - start_index
+
+        # Latitude: there might be slight numerical differences (-> check that
+        # the differences are very small before fixing it)
+        close_kwargs = {'rtol': 1e-3, 'atol': 1e-5}
+        if not np.allclose(
+                face_lat.bounds,
+                node_lat.points[conn_node_inds],
+                **close_kwargs,
+        ):
+            logger.warning(
+                "Latitude bounds of the face coordinate ('clat_vertices' in "
+                "the grid file) differ from the corresponding values "
+                "calculated from the connectivity ('vertex_of_cell') and the "
+                "node coordinate ('vlat'). Using bounds defined by "
+                "connectivity."
+            )
+        face_lat.bounds = node_lat.points[conn_node_inds]
+
+        # Longitude: there might be differences at the poles, where the
+        # longitude information does not matter (-> check that the only large
+        # differences are located at the poles). In addition, values might
+        # differ by 360°, which is also okay.
+        face_lon_bounds_to_check = face_lon.bounds % 360
+        node_lon_conn_to_check = node_lon.points[conn_node_inds] % 360
+        idx_notclose = ~np.isclose(
+            face_lon_bounds_to_check,
+            node_lon_conn_to_check,
+            **close_kwargs,
+        )
+        if not np.allclose(np.abs(face_lat.bounds[idx_notclose]), 90.0):
+            logger.warning(
+                "Longitude bounds of the face coordinate ('clon_vertices' in "
+                "the grid file) differ from the corresponding values "
+                "calculated from the connectivity ('vertex_of_cell') and the "
+                "node coordinate ('vlon'). Note that these values are allowed "
+                "to differ by 360° or at the poles of the grid. Using bounds "
+                "defined by connectivity."
+            )
+        face_lon.bounds = node_lon.points[conn_node_inds]
+
+        # Create mesh
+        connectivity = Connectivity(
+            indices=vertex_of_cell.data,
+            cf_role='face_node_connectivity',
+            start_index=start_index,
+            location_axis=0,
+        )
+        mesh = Mesh(
+            topology_dimension=2,
+            node_coords_and_axes=[(node_lat, 'y'), (node_lon, 'x')],
+            connectivities=[connectivity],
+            face_coords_and_axes=[(face_lat, 'y'), (face_lon, 'x')],
+        )
+
+        return mesh
+
+    def _get_grid_url(self, cube):
+        """Get ICON grid URL from cube."""
+        if self.GRID_FILE_ATTR not in cube.attributes:
```

Comment:
hope this is defined somewhere in the unchanged code in this PR, right?

Reply:
Yes, it is! https://github.com/ESMValGroup/ESMValCore/blob/a40c3a3ddeaed101bbf134bdec7ad220d71d7d1c/esmvalcore/cmor/_fixes/icon/_base_fixes.py#L28